lib/krypto/challenge.rb

```diff
@@ -57,12 +57,20 @@ def request_data
         @inner&.requestData
       end
 
-      def respond(signing_key, response_data)
+      # respond creates a reponse to the challenge. It accepts 2 keys to sign
+      # the response with, the second may be nil.
+      def respond(signing_key, signing_key_2, response_data)
         raise "No inner. Unverified?" unless @inner
 
+        public_signing_key_2_der = ""
+        if !signing_key_2.nil?
+          public_signing_key_2_der = Base64.strict_encode64(signing_key_2.public_to_der)
+        end
```

Comment:
```suggestion
        public_signing_key_2_der =  if !signing_key_2.nil?
    Base64.strict_encode64(signing_key_2.public_to_der)
    else
    ""
        end
```

is a bit more ruby. Or you can use a ternary. But the declare and the if declare is very go. :)

Reply:
nice, much cleaner!